samples/minimal/src/main.rs

```diff
@@ -52,30 +52,34 @@ impl MyFrame {
 
         frame.create_status_bar(2, wx::STB_DEFAULT_STYLE, 0, "statusBar");
         frame.set_status_text("Welcome to wxRust!", 0);
-        MyFrame { base: frame }
+        MyFrame {
+            base: frame.to_weak_ref(),
+        }
     }
 
     fn handle_menu(&self, event: &wx::CommandEvent) {
         println!("event={}", event.get_id());
-        match event.get_id() {
-            MINIMAL_QUIT => self.on_quit(),
-            MINIMAL_ABOUT => self.on_about(),
-            _ => (),
-        };
+        if let Some(frame) = self.base.get() {
+            match event.get_id() {
+                MINIMAL_QUIT => self.on_quit(&frame),
+                MINIMAL_ABOUT => self.on_about(&frame),
+                _ => (),
+            };
+        }
     }
 
-    fn on_quit(&self) {
-        self.base.close(true);
+    fn on_quit<F: FrameMethods>(&self, frame: &F) {
+        frame.close(true);
```

Comment:
Once typedef to WeakRef is defined, generic fn for readability is not needed anymore